objects/src/notes/vault.rs

```diff
@@ -108,3 +108,23 @@ impl NoteVault {
         padded_assets
     }
 }
+
+impl TryFrom<&[Word]> for NoteVault {
+    type Error = NoteError;
+
+    fn try_from(value: &[Word]) -> Result<Self, Self::Error> {
+        if value.is_empty() {
+            return Err(NoteError::EmptyAssetList);
+        } else if value.len() > Self::MAX_NUM_ASSETS {
+            return Err(NoteError::too_many_assets(value.len() / WORD_SIZE));
```

Comment:
I think this should be: `return Err(NoteError::too_many_assets(value.len()));`

Reply:
good catch!